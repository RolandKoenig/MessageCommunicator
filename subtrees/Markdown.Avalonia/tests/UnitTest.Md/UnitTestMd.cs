using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using UnitTest.Base;
using UnitTest.Base.Utils;

namespace UnitTest.Md
{
    [UseReporter(typeof(DiffReporter))]
    public class UnitTestMd : UnitTestBase
    {
        [Test]
        [RunOnUI]
        public void Transform_givenTest1_generatesExpectedResult()
        {
            var text = Util.LoadText("Test1.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists1_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists1.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists2_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists2.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLists3_generatesExpectedResult()
        {
            var text = Util.LoadText("Lists3.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenTables1_generatesExpectedResult()
        {
            var text = Util.LoadText("Tables.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenHorizontalRules_generatesExpectedResult()
        {
            var text = Util.LoadText("HorizontalRules.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenLinksInline_generatesExpectedResult()
        {
            var text = Util.LoadText("Links_inline_style.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenTextStyles_generatesExpectedResult()
        {
            var text = Util.LoadText("Text_style.md");
            var markdown = new Markdown.Avalonia.Markdown();
            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenImages_generatesExpectedResult()
        {
            var text = Util.LoadText("Images.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenBlockqoute_generatesExpectedResult()
        {
            var text = Util.LoadText("Blockquite.md");
            var markdown = new Markdown.Avalonia.Markdown();

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenMixing_generatesExpectedResult()
        {
            var text = Util.LoadText("Mixing.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenCodes_generatesExpectedResult()
        {
            var text = Util.LoadText("Codes.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }

        [Test]
        [RunOnUI]
        public void Transform_givenEmoji()
        {
            var text = Util.LoadText("Emoji.md");
            var markdown = new Markdown.Avalonia.Markdown();
            markdown.AssetPathRoot = AssetPath;

            var result = markdown.Transform(text);
            Approvals.Verify(Util.AsXaml(result));
        }
    }

}