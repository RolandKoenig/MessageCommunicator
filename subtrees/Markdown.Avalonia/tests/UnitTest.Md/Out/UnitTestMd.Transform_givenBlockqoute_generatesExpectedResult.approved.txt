<?xml version="1.0" encoding="utf-16"?>
<StackPanel xmlns:c="clr-namespace:ColorTextBlock.Avalonia;assembly=ColorTextBlock.Avalonia" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns="https://github.com/avaloniaui">
  <Border Classes="Blockquote" xmlns="">
    <Border.Child>
      <StackPanel Classes="Blockquote">
        <c:CTextBlock Classes="Heading2">
          <c:CRun Text="Features" />
        </c:CTextBlock>
        <c:CTextBlock Classes="Paragraph">
          <c:CRun Text="MarkDown.Xaml has a number of convenient features" />
        </c:CTextBlock>
        <Grid Classes="List">
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="0">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="The engine itself is a single file, for easy inclusion in your own projects" />
            </c:CTextBlock>
          </StackPanel>
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="1">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="1">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="Code for the engine is structured in the same manner as the original MarkdownSharp " />
            </c:CTextBlock>
          </StackPanel>
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="2">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="2">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="Includes a " />
              <c:CCode>
                <c:CRun Text="TextToFlowDocumentConverter" />
              </c:CCode>
              <c:CRun Text=" to make it easy to bind Markdown text" />
            </c:CTextBlock>
          </StackPanel>
          <Grid.ColumnDefinitions>
            <ColumnDefinition />
            <ColumnDefinition />
          </Grid.ColumnDefinitions>
          <Grid.RowDefinitions>
            <RowDefinition />
            <RowDefinition />
            <RowDefinition />
          </Grid.RowDefinitions>
        </Grid>
      </StackPanel>
    </Border.Child>
  </Border>
  <Border Classes="Blockquote" xmlns="">
    <Border.Child>
      <StackPanel Classes="Blockquote">
        <c:CTextBlock Classes="Heading2">
          <c:CRun Text="Features" />
        </c:CTextBlock>
        <c:CTextBlock Classes="Paragraph">
          <c:CRun Text="MarkDown.Xaml has a number of convenient features" />
        </c:CTextBlock>
        <Grid Classes="List">
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="0">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="0">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="The engine itself is a single file, for easy inclusion in your own projects" />
            </c:CTextBlock>
          </StackPanel>
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="1">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="1">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="Code for the engine is structured in the same manner as the original MarkdownSharp " />
            </c:CTextBlock>
          </StackPanel>
          <c:CTextBlock Classes="ListMarker" Grid.Column="0" Grid.Row="2">
            <c:CRun Text="•" />
          </c:CTextBlock>
          <StackPanel Grid.Column="1" Grid.Row="2">
            <c:CTextBlock Classes="Paragraph">
              <c:CRun Text="Includes a " />
              <c:CCode>
                <c:CRun Text="TextToFlowDocumentConverter" />
              </c:CCode>
              <c:CRun Text=" to make it easy to bind Markdown text" />
            </c:CTextBlock>
          </StackPanel>
          <Grid.ColumnDefinitions>
            <ColumnDefinition />
            <ColumnDefinition />
          </Grid.ColumnDefinitions>
          <Grid.RowDefinitions>
            <RowDefinition />
            <RowDefinition />
            <RowDefinition />
          </Grid.RowDefinitions>
        </Grid>
      </StackPanel>
    </Border.Child>
  </Border>
</StackPanel>
